diff -U 3 -H -d -r -N -- 2. First Hole/.project 1. Initial Tee/.project
--- 2. First Hole/.project	2012-06-08 00:19:30.000000000 +0200
+++ 1. Initial Tee/.project	2012-06-08 00:19:30.000000000 +0200
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>2. First Hole</name>
+	<name>1. Initial Tee</name>
 	<comment></comment>
 	<projects>
 	</projects>
diff -U 3 -H -d -r -N -- 2. First Hole/1. Initial Tee.iml 1. Initial Tee/1. Initial Tee.iml
--- 2. First Hole/1. Initial Tee.iml	1970-01-01 01:00:00.000000000 +0100
+++ 1. Initial Tee/1. Initial Tee.iml	2012-10-06 13:57:27.847273891 +0200
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="false">
+    <output url="file://$MODULE_DIR$/bin" />
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src/main" isTestSource="false" />
+      <sourceFolder url="file://$MODULE_DIR$/src/test" isTestSource="false" />
+    </content>
+    <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="jdk" jdkName="1.7" jdkType="JavaSDK" />
+    <orderEntry type="module-library">
+      <library name="junit4">
+        <CLASSES>
+          <root url="jar://$APPLICATION_HOME_DIR$/lib/junit-4.10.jar!/" />
+        </CLASSES>
+        <JAVADOC />
+        <SOURCES />
+      </library>
+    </orderEntry>
+  </component>
+</module>
+
diff -U 3 -H -d -r -N -- 2. First Hole/2. First Hole.iml 1. Initial Tee/2. First Hole.iml
--- 2. First Hole/2. First Hole.iml	2012-10-06 13:57:27.855273891 +0200
+++ 1. Initial Tee/2. First Hole.iml	1970-01-01 01:00:00.000000000 +0100
@@ -1,23 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<module type="JAVA_MODULE" version="4">
-  <component name="NewModuleRootManager" inherit-compiler-output="false">
-    <output url="file://$MODULE_DIR$/bin" />
-    <exclude-output />
-    <content url="file://$MODULE_DIR$">
-      <sourceFolder url="file://$MODULE_DIR$/src/main" isTestSource="false" />
-      <sourceFolder url="file://$MODULE_DIR$/src/test" isTestSource="false" />
-    </content>
-    <orderEntry type="sourceFolder" forTests="false" />
-    <orderEntry type="jdk" jdkName="1.7" jdkType="JavaSDK" />
-    <orderEntry type="module-library">
-      <library name="junit4">
-        <CLASSES>
-          <root url="jar://$APPLICATION_HOME_DIR$/lib/junit-4.10.jar!/" />
-        </CLASSES>
-        <JAVADOC />
-        <SOURCES />
-      </library>
-    </orderEntry>
-  </component>
-</module>
-
Binary files 2. First Hole/bin/refactoringgolf/store1/Customer.class and 1. Initial Tee/bin/refactoringgolf/store1/Customer.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/CustomerTest.class and 1. Initial Tee/bin/refactoringgolf/store1/CustomerTest.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/Employee.class and 1. Initial Tee/bin/refactoringgolf/store1/Employee.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/ImageInfo.class and 1. Initial Tee/bin/refactoringgolf/store1/ImageInfo.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/Manager.class and 1. Initial Tee/bin/refactoringgolf/store1/Manager.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/ManagerTest.class and 1. Initial Tee/bin/refactoringgolf/store1/ManagerTest.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/Order.class and 1. Initial Tee/bin/refactoringgolf/store1/Order.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/OrderItem.class and 1. Initial Tee/bin/refactoringgolf/store1/OrderItem.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/OrderTest.class and 1. Initial Tee/bin/refactoringgolf/store1/OrderTest.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/Product.class and 1. Initial Tee/bin/refactoringgolf/store1/Product.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/ProductCategory.class and 1. Initial Tee/bin/refactoringgolf/store1/ProductCategory.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/ProductTest.class and 1. Initial Tee/bin/refactoringgolf/store1/ProductTest.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/Salesman.class and 1. Initial Tee/bin/refactoringgolf/store1/Salesman.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/SalesmanTest.class and 1. Initial Tee/bin/refactoringgolf/store1/SalesmanTest.class differ
Binary files 2. First Hole/bin/refactoringgolf/store1/ThirdParty.class and 1. Initial Tee/bin/refactoringgolf/store1/ThirdParty.class differ
diff -U 3 -H -d -r -N -- 2. First Hole/src/main/refactoringgolf/store/ClassDiagram.ucls 1. Initial Tee/src/main/refactoringgolf/store/ClassDiagram.ucls
--- 2. First Hole/src/main/refactoringgolf/store/ClassDiagram.ucls	2012-06-08 00:19:30.000000000 +0200
+++ 1. Initial Tee/src/main/refactoringgolf/store/ClassDiagram.ucls	2012-06-08 00:19:30.000000000 +0200
@@ -1,13 +1,87 @@
 <class-diagram version="1.0.4" icons="true" always-add-relationships="false" generalizations="true" realizations="true" 
   associations="true" dependencies="false" nesting-relationships="true">  
-  <class id="1" corner="BOTTOM_RIGHT" language="java" name="refactoringgolf.store.Order" project="2. First Hole" 
-    file="/2. First Hole/src/main/refactoringgolf/store/Order.java" binary="false">    
-    <position x="40" y="10" width="272" height="355"/>    
+  <class id="1" corner="BOTTOM_RIGHT" language="java" name="refactoringgolf.store.Order" project="1. Initial Tee" 
+    file="/1. Initial Tee/src/main/refactoringgolf/store/Order.java" binary="false">    
+    <position x="158" y="154" width="-1" height="-1"/>    
+    <display autosize="true" package="true" initial-value="false" signature="true" visibility="true">      
+      <attributes public="true" package="true" protected="true" private="true"/>      
+      <operations public="true" package="true" protected="true" private="true"/>    
+    </display>  
+  </class>  
+  <class id="2" corner="BOTTOM_RIGHT" language="java" name="refactoringgolf.store.OrderItem" project="1. Initial Tee" 
+    file="/1. Initial Tee/src/main/refactoringgolf/store/OrderItem.java" binary="false">    
+    <position x="454" y="153" width="-1" height="-1"/>    
+    <display autosize="true" package="true" initial-value="false" signature="true" visibility="true">      
+      <attributes public="true" package="true" protected="true" private="true"/>      
+      <operations public="true" package="true" protected="true" private="true"/>    
+    </display>  
+  </class>  
+  <class id="3" corner="BOTTOM_RIGHT" language="java" name="refactoringgolf.store.Employee" project="1. Initial Tee" 
+    file="/1. Initial Tee/src/main/refactoringgolf/store/Employee.java" binary="false">    
+    <position x="148" y="505" width="-1" height="-1"/>    
+    <display autosize="true" package="true" initial-value="false" signature="true" visibility="true">      
+      <attributes public="true" package="true" protected="true" private="true"/>      
+      <operations public="true" package="true" protected="true" private="true"/>    
+    </display>  
+  </class>  
+  <class id="4" corner="BOTTOM_RIGHT" language="java" name="refactoringgolf.store.Manager" project="1. Initial Tee" 
+    file="/1. Initial Tee/src/main/refactoringgolf/store/Manager.java" binary="false">    
+    <position x="370" y="348" width="229" height="99"/>    
     <display autosize="true" package="true" initial-value="false" signature="true" visibility="true">      
       <attributes public="true" package="true" protected="true" private="true"/>      
       <operations public="true" package="true" protected="true" private="true"/>    
     </display>  
   </class>  
+  <class id="5" corner="BOTTOM_RIGHT" language="java" name="refactoringgolf.store.Salesman" project="1. Initial Tee" 
+    file="/1. Initial Tee/src/main/refactoringgolf/store/Salesman.java" binary="false">    
+    <position x="461" y="586" width="-1" height="-1"/>    
+    <display autosize="true" package="true" initial-value="false" signature="true" visibility="true">      
+      <attributes public="true" package="true" protected="true" private="true"/>      
+      <operations public="true" package="true" protected="true" private="true"/>    
+    </display>  
+  </class>  
+  <generalization>    
+    <end type="SOURCE" nodeId="4"/>    
+    <end type="TARGET" nodeId="3"/>  
+  </generalization>  
+  <association>    
+    <end type="SOURCE" nodeId="1" navigable="false">      
+      <attribute name="items"/>      
+      <multiplicity minimum="0" maximum="2147483647"/>    
+    </end>    
+    <end type="TARGET" nodeId="2" navigable="true"/>    
+    <display labels="true" multiplicity="true"/>  
+  </association>  
+  <generalization>    
+    <end type="SOURCE" nodeId="5"/>    
+    <end type="TARGET" nodeId="3"/>  
+  </generalization>  
+  <association>    
+    <end type="SOURCE" nodeId="3" navigable="false">      
+      <attribute name="subordinates">        
+        <position x="278" y="677" width="69" height="14"/>      
+      </attribute>      
+      <multiplicity minimum="0" maximum="2147483647">        
+        <position x="259" y="676" width="16" height="14"/>      
+      </multiplicity>    
+    </end>    
+    <end type="TARGET" nodeId="3" navigable="true"/>    
+    <display labels="true" multiplicity="true"/>  
+  </association>  
+  <association>    
+    <bendpoint x="147" y="711"/>    
+    <bendpoint x="160" y="711"/>    
+    <end type="SOURCE" nodeId="3" navigable="false">      
+      <attribute name="manager">        
+        <position x="158" y="712" width="48" height="14"/>      
+      </attribute>      
+      <multiplicity minimum="0" maximum="1">        
+        <position x="133" y="713" width="18" height="14"/>      
+      </multiplicity>    
+    </end>    
+    <end type="TARGET" nodeId="3" navigable="true"/>    
+    <display labels="true" multiplicity="true"/>  
+  </association>  
   <classifier-display autosize="true" package="true" initial-value="false" signature="true" visibility="true">    
     <attributes public="true" package="true" protected="true" private="true"/>    
     <operations public="true" package="true" protected="true" private="true"/>  
diff -U 3 -H -d -r -N -- 2. First Hole/src/main/refactoringgolf/store/Order.java 1. Initial Tee/src/main/refactoringgolf/store/Order.java
--- 2. First Hole/src/main/refactoringgolf/store/Order.java	2012-10-06 14:23:09.823309075 +0200
+++ 1. Initial Tee/src/main/refactoringgolf/store/Order.java	2012-10-06 14:22:00.143307484 +0200
@@ -52,52 +52,39 @@
 		return items;
 	}
 
+    // BEGIN 1. CHANGE
+    // Split into shipping-tax-totalItems
 	public float total() {
-		float totalItems = totalItems();
-		float tax = tax(totalItems);
-		int shipping = shipping();
-
-		return totalItems + tax + shipping;
-	}
-
-	private int shipping() {
-		int shipping = 15;
-		if (this.deliveryCountry == "USA") {
-			shipping = 0;
-		}
-		return shipping;
-	}
-	
-	private float tax(float totalAmount) {
-		return totalAmount * 5 / 100;
-	}
-
-	private float totalItems() {
-		float totalAmount = 0;
+		float totalItems = 0;
 		for (OrderItem item : items) {
-            // BEGIN 2. CHANGE
-            // call functions on orderItem
-			float discount=0;
 			float itemAmount = item.getProduct().getUnitPrice() * item.getQuantity();
 			if (item.getProduct().getCategory() == ProductCategory.Accessories) {
-				discount = 0;
+				float booksDiscount = 0;
 				if (itemAmount >= 100) {
-					discount = itemAmount * 10 / 100;
+					booksDiscount = itemAmount * 10 / 100;
 				}
+				itemAmount = itemAmount - booksDiscount;
 			}
 			if (item.getProduct().getCategory() == ProductCategory.Bikes) {
-				discount = itemAmount * 20 / 100;
+				// itemAmount=itemAmount-discount
+				itemAmount = itemAmount - itemAmount * 20 / 100;
 			}
 			if (item.getProduct().getCategory() == ProductCategory.Cloathing) {
-				discount = 0;
+				float cloathingDiscount = 0;
 				if (item.getQuantity() > 2) {
-					discount = item.getProduct().getUnitPrice();
+					cloathingDiscount = item.getProduct().getUnitPrice();
 				}
+				itemAmount = itemAmount - cloathingDiscount;
 			}
-			itemAmount = itemAmount - discount;
-			totalAmount += itemAmount;
-            // END 2. CHANGE
+			totalItems += itemAmount;
 		}
-		return totalAmount;
+
+		if (this.deliveryCountry == "USA")
+			// totalAmount=totalItemAmount + tax + 0 shipping
+			return totalItems + totalItems * 5 / 100;
+
+		// totalAmount=totalItemAmount + tax + 15 shipping
+		return totalItems + totalItems * 5 / 100 + 15;
 	}
+    // END 1. CHANGE
 }
diff -U 3 -H -d -r -N -- 2. First Hole/src/main/refactoringgolf/store/OrderItem.java 1. Initial Tee/src/main/refactoringgolf/store/OrderItem.java
--- 2. First Hole/src/main/refactoringgolf/store/OrderItem.java	2012-10-06 14:13:29.267295828 +0200
+++ 1. Initial Tee/src/main/refactoringgolf/store/OrderItem.java	2012-10-06 14:24:23.543310756 +0200
@@ -20,4 +20,9 @@
 	public int getQuantity() {
 		return quantity;
 	}
+
+    // BEGIN 2. CHANGE
+    // new function total ... called from Order
+
+    // END 2. CHANGE
 }
diff -U 3 -H -d -r -N -- 2. First Hole/src/main/refactoringgolf/store/Salesman.java 1. Initial Tee/src/main/refactoringgolf/store/Salesman.java
--- 2. First Hole/src/main/refactoringgolf/store/Salesman.java	2012-10-06 14:13:29.235295827 +0200
+++ 1. Initial Tee/src/main/refactoringgolf/store/Salesman.java	2012-10-06 14:13:29.271295828 +0200
@@ -5,10 +5,10 @@
 	private int commissionPorcentage;
 	private float monthQuota;
 
-	public Salesman(String firstName, String lastName, float fixedSalary, int commissionPorcentage) {
-		super(firstName, lastName, fixedSalary);
-		this.commissionPorcentage = commissionPorcentage;
-	}
+	public Salesman(String firstName, String lastName, float fixedSalary, int commissionPorcentage){
+    	super(firstName, lastName, fixedSalary);
+        this.commissionPorcentage = commissionPorcentage;
+    }
 
 	public void setCommissionPorcentage(int commissionPorcentage) {
 		this.commissionPorcentage = commissionPorcentage;
@@ -28,7 +28,7 @@
 		float tax = 0;
 		if (fixedSalary > 3500)
 			tax = fixedSalary * 5 / 100;
-		return fixedSalary + benefits - pensionFounds - tax;
+		return fixedSalary+benefits - pensionFounds - tax;
 	}
 
 	public void updateMonthQuota(float addQuota) {
